### PR TITLE
feat: database_schema support in dust apps

### PR DESCRIPTION
**Pull request by fontanierh**
It works:

Screenshot 2023-11-22 at 17 32 07

Screenshot 2023-11-22 at 17 32 16

I think I didn't miss any place 🤞

### CODE REVIEW

File: front/components/app/blocks/DatabaseSchema.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                  workspace_id: ds.workspace_id,
+                  data_source_id: ds.data_source_id,
+                };
+                onBlockUpdate(b);
+              }}
+            />
+          </div>
+        </div>
+        {block.config.database?.data_source_id && (
+          <div className="flex flex-col xl:flex-row xl:space-x-2">
```
Comment (spolu) [inline front/components/app/blocks/DatabaseSchema.tsx:93]: can we try to put them on the same line on wide screen (and vertical alignment on small screens?

File: front/components/app/NewBlock.tsx [hunk 2]
```diff
@@ -164,7 +170,7 @@ export default function NewBlock({
                     ))}
                   </div>
                 </div>
-                <div className="col-span-8 pr-2 text-sm text-gray-700 sm:col-span-9 sm:pl-3">
+                <div className="col-span-8 pr-2 text-sm text-gray-700 sm:col-span-9 sm:pl-6">
```
Comment (spolu) [inline front/components/app/NewBlock.tsx:173]: This is because `database_schema` was too long?

File: front/lib/config.ts [hunk 3]
```diff
@@ -90,6 +91,27 @@ export function extractConfig(spec: SpecificationType): BlockRunConfig {
             : false,
         };
         break;
+      case "database_schema":
+        console.log("\n\n\n\nHELLLOW:", spec[i].config);
```
Comment (spolu) [inline front/lib/config.ts:95]: console.log

File: front/components/database/DatabasePicker.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                  <>
+                    <div className="mr-1 text-sm font-bold text-action-500">
+                      {currentDatabase.name}
+                    </div>
+                    <ChevronDownIcon className="mt-0.5 h-4 w-4 hover:text-gray-700" />
+                  </>
+                ) : databases && databases.length > 0 ? (
+                  "Select Database"
+                ) : (
+                  "There are no databases in this data source."
```
Comment (spolu) [inline front/components/database/DatabasePicker.tsx:75]: THat's button-ish UI right? should this be "No Databases" ?

**Review by spolu — COMMENTED** [review]
Looks good left a few minor comments

File: front/components/app/NewBlock.tsx (hunk 2 above)
Comment (fontanierh) [reply front/components/app/NewBlock.tsx:173]: yup

File: front/lib/config.ts (hunk 3 above)
Comment (fontanierh) [reply front/lib/config.ts:95]: oups missed this one thx 🙏

File: front/components/database/DatabasePicker.tsx (hunk 4 above)
Comment (fontanierh) [reply front/components/database/DatabasePicker.tsx:75]: It is button-ish indeed, will change 👍

File: front/components/app/blocks/DatabaseSchema.tsx (hunk 1 above)
Comment (fontanierh) [reply front/components/app/blocks/DatabaseSchema.tsx:95]: Fixed

**Review by spolu — DISMISSED** [review]

**Pull request merged**